### PR TITLE
epee: bump default timeout, and ease it up for all local IPs

### PR DESCRIPTION
**Pull request by moneromooo-monero**
also fix time/timeout typo, it's pretty confusing otherwise

### CODE REVIEW

File: contrib/epee/include/net/abstract_tcp_server2.inl [hunk 1]
```diff
@@ -146,7 +146,7 @@ PRAGMA_WARNING_DISABLE_VS(4355)
 
     context = boost::value_initialized<t_connection_context>();
     const unsigned long ip_{boost::asio::detail::socket_ops::host_to_network_long(remote_ep.address().to_v4().to_ulong())};
-    m_local = epee::net_utils::is_ip_loopback(ip_);
+    m_local = epee::net_utils::is_ip_local(ip_);
```
Comment (stoffu) [inline contrib/epee/include/net/abstract_tcp_server2.inl]: What does this change entail? Shouldn't this be a separate commit for better visibility?

Comment (moneromooo-monero) [reply contrib/epee/include/net/abstract_tcp_server2.inl]: It allows non routable IPs to use the longer timeout. It could be a separate commit, I just didn't see enough of a need since it's related to the same thing, but it's true it's a separate unit.
